### PR TITLE
fixes issues #9 and #10

### LINKED ISSUE #9: Deleted products not reflecting on UI
After deleting an item in the product fragment, the changes don't reflect on the screen when you return to the shopping list screen. Occurs when the shopping list item is the first on the screen

### LINKED ISSUE #10: Data sort operations in intent processor
Noticed some sorting functions called in the intent processor. Believe its best they exist in the data layer

### PR DESCRIPTION
**Pull request by Ezike**
fixes  #9 
Updated the diff util in shopping list adapter to take note of product list size in order to reflect the changes always
fixes #10

### CODE REVIEW

File: shopping_list/src/main/java/com/zizohanto/android/tobuy/shopping_list/ui/shopping_list/adaper/ShoppingListAdapter.kt [hunk 1]
```diff
@@ -100,7 +99,8 @@ class ShoppingListAdapter @Inject constructor() :
                     oldItem: ShoppingListWithProductsModel,
                     newItem: ShoppingListWithProductsModel
                 ): Boolean {
-                    return oldItem.shoppingList == newItem.shoppingList
+                    return oldItem.products.size == newItem.products.size &&
```
Comment (zizoh) [inline shopping_list/src/main/java/com/zizohanto/android/tobuy/shopping_list/ui/shopping_list/adaper/ShoppingListAdapter.kt:102]: Why are we also checking the size of the products?

Comment (tobenna-paystack) [reply shopping_list/src/main/java/com/zizohanto/android/tobuy/shopping_list/ui/shopping_list/adaper/ShoppingListAdapter.kt:102]: that's the fix

Comment (Ezike) [reply shopping_list/src/main/java/com/zizohanto/android/tobuy/shopping_list/ui/shopping_list/adaper/ShoppingListAdapter.kt:102]: that fixes the issue linked to the PR

Comment (zizoh) [reply shopping_list/src/main/java/com/zizohanto/android/tobuy/shopping_list/ui/shopping_list/adaper/ShoppingListAdapter.kt:102]: Oh, seen.

**Pull request merged**